test/enforce-compiletime-test.cc

```diff
@@ -0,0 +1,137 @@
+// Formatting library for C++ - formatting library tests
+//
+// Copyright (c) 2012 - present, Victor Zverovich
+// All rights reserved.
+//
+// For the license information refer to format.h.
+
+#include <array>
+#include <iterator>
+#include <list>
+#include <map>
+#include <sstream>
+#include <string>
+#include <utility>
+#include <vector>
+
+#ifdef WIN32
+#  define _CRT_SECURE_NO_WARNINGS
+#endif
+
+#include "fmt/chrono.h"
+#include "fmt/color.h"
+#include "fmt/format.h"
+#include "fmt/locale.h"
+#include "fmt/ostream.h"
+#include "fmt/ranges.h"
+
+// Exercise the API to verify that everything we expect to can compile.
+void TestFormatApi() {
+  (void)fmt::format(FMT_STRING("{}"), 42);
+  (void)fmt::format(FMT_STRING(L"{}"), 42);
+#if !FMT_GCC_VERSION
+  (void)fmt::format(FMT_STRING("noop"));
+#endif
+
+  (void)fmt::to_string(42);
+  (void)fmt::to_wstring(42);
+
+  std::list<char> out;
+  fmt::format_to(std::back_inserter(out), FMT_STRING("{}"), 42);
+  std::stringstream s;
+  fmt::format_to(std::ostream_iterator<char>(s), FMT_STRING("{}"), 42);
+
+  char buffer[4];
+  (void)fmt::format_to_n(buffer, 3, FMT_STRING("{}"), 12345);
+
+  wchar_t wbuffer[4];
+  (void)fmt::format_to_n(wbuffer, 3, FMT_STRING(L"{}"), 12345);
+
+  (void)fmt::formatted_size(FMT_STRING("{}"), 42);
+}
+void TestLiteralsApi() {
+#if FMT_USE_UDL_TEMPLATE
+  // Passing user-defined literals directly to EXPECT_EQ causes problems
+  // with macro argument stringification (#) on some versions of GCC.
+  // Workaround: Assing the UDL result to a variable before the macro.
+
+  using namespace fmt::literals;
+
+  auto udl_format = "{}c{}"_format("ab", 1);
+  auto udl_format_w = L"{}c{}"_format(L"ab", 1);
+  (void)udl_format;
+  (void)udl_format_w;
+#endif
+}
+
+struct test_output_iterator {
+  char* data;
+
+  using iterator_category = std::output_iterator_tag;
+  using value_type = void;
+  using difference_type = void;
+  using pointer = void;
+  using reference = void;
+
+  test_output_iterator& operator++() {
+    ++data;
+    return *this;
+  }
+  test_output_iterator operator++(int) {
+    auto tmp = *this;
+    ++data;
+    return tmp;
+  }
+  char& operator*() { return *data; }
+};
+
+void FormatToNOutputIteratorTest() {
+  char buf[10] = {};
+  fmt::format_to_n(test_output_iterator{buf}, 10, FMT_STRING("{}"), 42);
+}
+
+void TestChrono() {
+#ifndef FMT_STATIC_THOUSANDS_SEPARATOR
```

Comment:
Why conditionally compiled?

Reply:
TBH, I'm not sure - all the tests covering formatting of chrono types in chrono_test.cc were too so I figured it was important? I'll try removing it

Reply:
I lack context for this particular test, but: `FMT_STATIC_THOUSANDS_SEPARATOR` is "overloaded" to mean "I don't like `<locale>` and want nothing to do with it", and so the `locale_ref` functions are not implemented if `FMT_STATIC_THOUSANDS_SEPARATOR` is defined. However, `chrono_formatter::format_localized` always tries to use the `locale_ref` functions, so `chrono-test` won't link in that situation.

Reply:
Thank you for the explanation. It seems to work just fine without the ifdef in this situation so I'll leave them removed.